src/Core.tsx

```diff
@@ -417,12 +416,14 @@ export class Core extends React.Component<IModalProps, IModalState> {
      const did = this.did()
 
      if (!backendUrl) {
+       this.setState({ show: false })
        return onConnect(provider, this.disconnect, this.selectedLanguageCode, this.selectedTheme, dataVault)
      }
 
      const handleConnect = (provider: any, authKeys: AuthKeys) => onConnect(provider, this.disconnect, this.selectedLanguageCode, this.selectedTheme, dataVault, authKeys)
 
      return confirmAuth(provider, address!, backendUrl!, did, challenge!, handleConnect, sd)
+       .then(() => this.setState({ show: false }))
```

Comment:
This is missing in the _retry_. See that we are doing `confirmAuth` again if something failed

Reply:
Isn't that what the catch right below it is doing? Attempting to reconnect?